### PR TITLE
Add new flash attention forward for inference

### PR DESCRIPTION
**Pull request by ltqin**

### CODE REVIEW

File: include/ck/tensor_operation/gpu/device/device_batched_multiple_head_flash_attention_fwd.hpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+          typename CDataType,
+          typename Acc0BiasDataType,
+          typename Acc1BiasDataType,
+          typename AElementwiseOperation,
+          typename B0ElementwiseOperation,
+          typename C0ElementwiseOperation,
+          typename B1ElementwiseOperation,
+          typename C1DEElementwiseOperation,
+          MaskingSpecialization MaskingSpec>
+struct DeviceBatchedMultiheadAttentionForward : public BaseOperator
```
Comment (qianfengz) [inline include/ck/tensor_operation/gpu/device/device_batched_multiple_head_flash_attention_fwd.hpp]: Rename the class name to `DeviceBatchedMultiheadAttentionInfer`

File: include/ck/tensor_operation/gpu/device/device_grouped_multiple_head_flash_attention_fwd.hpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+          typename CDataType,
+          typename Acc0BiasDataType,
+          typename Acc1BiasDataType,
+          typename AElementwiseOperation,
+          typename B0ElementwiseOperation,
+          typename Acc0ElementwiseOperation,
+          typename B1ElementwiseOperation,
+          typename CElementwiseOperation,
+          MaskingSpecialization MaskingSpec>
+struct DeviceGroupedMultiheadAttentionForward : public BaseOperator
```
Comment (qianfengz) [inline include/ck/tensor_operation/gpu/device/device_grouped_multiple_head_flash_attention_fwd.hpp]: Rename the class name to `DeviceGroupedMultiheadAttentionInfer`

**Review by qianfengz — COMMENTED** [review]
Rename the base class file name to 
`device_batched_mha_infer.hpp` and `device_grouped_mha_infer.hpp` respectively

**Review by qianfengz — COMMENTED** [review]
Rename the file name `device_batched_mha_fwd_xdl_cshuffle.hpp` to `device_batched_mha_infer_xdl_cshuffle.hpp`

**Review by qianfengz — COMMENTED** [review]
Rename the file name `device_grouped_mha_fwd_xdl_cshuffle.hpp` to `device_grouped_mha_infer_cshuffle.hpp`

**Review by qianfengz — COMMENTED** [review]
Rename the kernel file name `gridwise_batched_mha_fwd_xdl_cshuffle.hpp` to `gridwise_batched_mha_infer_xdl_cshuffle.hpp`

File: include/ck/tensor_operation/gpu/grid/gridwise_batched_mha_fwd_xdl_cshuffle.hpp [hunk 3]
```diff
@@ -0,0 +1,1261 @@
...
+          index_t B1BlockLdsExtraN,
+          index_t CShuffleMXdlPerWavePerShuffle,
+          index_t CShuffleNXdlPerWavePerShuffle,
+          typename CShuffleBlockTransferClusterLengths_MBlock_MPerBlock_NBlock_NPerBlock,
+          index_t CShuffleBlockTransferScalarPerVector_NPerBlock,
+          LoopScheduler LoopSched,
+          bool PadN,
+          bool MaskOutUpperTriangle,
+          PipelineVersion PipelineVer = PipelineVersion::v1>
+struct GridwiseMultiHeadFlashAttentionForward_Xdl_CShuffle
```
Comment (qianfengz) [inline include/ck/tensor_operation/gpu/grid/gridwise_batched_mha_fwd_xdl_cshuffle.hpp]: Rename to `GridwiseMultiHeadFlashAttentionInfer_Xdl_CShuffle`

File: include/ck/tensor_operation/gpu/device/impl/device_batched_mha_fwd_xdl_cshuffle.hpp [hunk 4]
```diff
@@ -0,0 +1,954 @@
...
+          typename D0GridDescriptor_M0_N0_M1_N1_M2_N2_M3_N3_N4_N5,
+          typename Block2CTileMap,
+          typename ComputeBasePtrOfStridedBatch,
+          typename C0MatrixMask,
+          bool HasMainKBlockLoop>
+__global__ void
+#if CK_USE_LAUNCH_BOUNDS
+    __launch_bounds__(CK_MAX_THREAD_PER_BLOCK, CK_MIN_BLOCK_PER_CU)
+#endif
+        kernel_batched_multiple_head_flash_attention_forward(
```
Comment (qianfengz) [inline include/ck/tensor_operation/gpu/device/impl/device_batched_mha_fwd_xdl_cshuffle.hpp]: Rename to `kernel_batched_multiple_head_flash_attention_infer`

File: include/ck/tensor_operation/gpu/device/impl/device_batched_mha_fwd_xdl_cshuffle.hpp [hunk 5]
```diff
@@ -0,0 +1,954 @@
...
+          index_t B1BlockTransferDstScalarPerVector_BK1,
+          bool B1BlockLdsExtraN,
+          index_t CShuffleMXdlPerWavePerShuffle,
+          index_t CShuffleNXdlPerWavePerShuffle,
+          typename CShuffleBlockTransferClusterLengths_MBlock_MPerBlock_NBlock_NPerBlock,
+          index_t CShuffleBlockTransferScalarPerVector_NPerBlock,
+          MaskingSpecialization MaskingSpec,
+          int D0sTransferSrcScalarPerVector = 4,
+          LoopScheduler LoopSched           = LoopScheduler::Default>
+struct DeviceBatchedMultiheadAttentionForward_Xdl
```
Comment (qianfengz) [inline include/ck/tensor_operation/gpu/device/impl/device_batched_mha_fwd_xdl_cshuffle.hpp]: Rename to `DeviceBatchedMultiheadAttentionInfer_Xdl`

File: include/ck/tensor_operation/gpu/device/impl/device_grouped_mha_fwd_xdl_cshuffle.hpp [hunk 6]
```diff
@@ -0,0 +1,987 @@
...
+          typename BElementwiseOperation,
+          typename AccElementwiseOperation,
+          typename B1ElementwiseOperation,
+          typename CElementwiseOperation,
+          bool HasMainKBlockLoop>
+__global__ void
+#if CK_USE_LAUNCH_BOUNDS
+    __launch_bounds__(CK_MAX_THREAD_PER_BLOCK, CK_MIN_BLOCK_PER_CU)
+#endif
+        kernel_grouped_gemm_softmax_gemm_xdl_cshuffle_v1(
```
Comment (qianfengz) [inline include/ck/tensor_operation/gpu/device/impl/device_grouped_mha_fwd_xdl_cshuffle.hpp]: Rename to `kernel_grouped_multiple_head_flash_attention_infer`

File: include/ck/tensor_operation/gpu/device/impl/device_grouped_mha_fwd_xdl_cshuffle.hpp [hunk 7]
```diff
@@ -0,0 +1,987 @@
...
+          index_t B1BlockTransferSrcScalarPerVector,
+          index_t B1BlockTransferDstScalarPerVector_BK1,
+          bool B1BlockLdsExtraN,
+          index_t CShuffleMXdlPerWavePerShuffle,
+          index_t CShuffleNXdlPerWavePerShuffle,
+          typename CShuffleBlockTransferClusterLengths_MBlock_MPerBlock_NBlock_NPerBlock,
+          index_t CShuffleBlockTransferScalarPerVector_NPerBlock,
+          MaskingSpecialization MaskingSpec,
+          LoopScheduler LoopSched = LoopScheduler::Default>
+struct DeviceGroupedMultiheadAttentionForward_Xdl
```
Comment (qianfengz) [inline include/ck/tensor_operation/gpu/device/impl/device_grouped_mha_fwd_xdl_cshuffle.hpp]: Rename to `DeviceGroupedMultiheadAttentionInfer_Xdl`

**Review by qianfengz — COMMENTED** [review]
Rename the file `batched_gemm_multihead_attention_bias_forward.cpp`  to `batched_gemm_multihead_attention_bias_infer.cpp`

**Review by qianfengz — COMMENTED** [review]
Rename example file `run_batched_multihead_attention_bias_forward.inc` to `run_batched_multihead_attention_bias_infer.inc`

**Review by qianfengz — COMMENTED** [review]
Rename example file `batched_gemm_multihead_attention_forward.cpp` to `batched_gemm_multihead_attention_infer.cpp`

**Review by qianfengz — COMMENTED** [review]
Rename example file `run_grouped_multihead_attention_bias_forward.inc` to `run_grouped_multihead_attention_bias_infer.inc`

File: include/ck/tensor_operation/gpu/device/device_batched_multiple_head_flash_attention_fwd.hpp (hunk 1 above)
Comment (ltqin) [reply include/ck/tensor_operation/gpu/device/device_batched_multiple_head_flash_attention_fwd.hpp]: done

File: include/ck/tensor_operation/gpu/device/device_grouped_multiple_head_flash_attention_fwd.hpp (hunk 2 above)
Comment (ltqin) [reply include/ck/tensor_operation/gpu/device/device_grouped_multiple_head_flash_attention_fwd.hpp]: done

File: include/ck/tensor_operation/gpu/grid/gridwise_batched_mha_fwd_xdl_cshuffle.hpp (hunk 3 above)
Comment (ltqin) [reply include/ck/tensor_operation/gpu/grid/gridwise_batched_mha_fwd_xdl_cshuffle.hpp]: done

File: include/ck/tensor_operation/gpu/device/impl/device_batched_mha_fwd_xdl_cshuffle.hpp (hunk 4 above)
Comment (ltqin) [reply include/ck/tensor_operation/gpu/device/impl/device_batched_mha_fwd_xdl_cshuffle.hpp]: done

File: include/ck/tensor_operation/gpu/device/impl/device_batched_mha_fwd_xdl_cshuffle.hpp (hunk 5 above)
Comment (ltqin) [reply include/ck/tensor_operation/gpu/device/impl/device_batched_mha_fwd_xdl_cshuffle.hpp]: done

File: include/ck/tensor_operation/gpu/device/impl/device_grouped_mha_fwd_xdl_cshuffle.hpp (hunk 6 above)
Comment (ltqin) [reply include/ck/tensor_operation/gpu/device/impl/device_grouped_mha_fwd_xdl_cshuffle.hpp]: done

File: include/ck/tensor_operation/gpu/device/impl/device_grouped_mha_fwd_xdl_cshuffle.hpp (hunk 7 above)
Comment (ltqin) [reply include/ck/tensor_operation/gpu/device/impl/device_grouped_mha_fwd_xdl_cshuffle.hpp]: done

**Review by qianfengz — APPROVED** [review]

**Pull request merged**